platforms/documentation/docs/src/docs/userguide/best-practices/bp_general.adoc

```diff
@@ -93,3 +93,66 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
 
 === Tags
 `#upgrades`
+
+[[favor_composite_builds]]
+== Favor Composite Builds over `buildSrc`
+:keywords: structuring-builds,composite-builds,buildSrc
+
+You should setup a <<composite_builds.adoc#defining_composite_builds,Composite Build>> (commonly called an "included build") to hold your _build logic_ (such as any plugins - including convention plugins - and other customizations that are specific to your build).
+
+=== Explanation
+
+The preferred location for build logic is an included build named `build-logic`, **not** in `buildSrc`.
+
+The automatically available `buildSrc` is great for rapid prototyping, but suffers from some subtle disadvantages:
+- There are classloader differences in how these 2 approaches behave that can be surprising; included builds are treated just like external dependencies, which is a simpler mental model.  Dependency resolution behaves subtly differently in buildSrc.
+- There can potentially be fewer task invalidations in a build when files in an included build are modified, leading to faster builds.
```

Comment:
I think we need to specify more details on when you get no-invalidations in included builds but you do in buildSrc. As-is, it is a bit vague.

Reply:
https://docs.gradle.org/current/userguide/organizing_gradle_projects.html#sec:build_sources says "A change in buildSrc causes the whole project to become out-of-date."  We could include that sentence here.

@wolfs - Is this still true?

@lkasso - If we do accept this item, should we change the linked section as part of this PR as well?  It might seem a little at odds with this item.

Reply:
I depends on what "out-of-date" means. All tasks that are not built-in Gradle tasks will be out of date. Though the same is true for changes in included plugin builds.

Reply:
> If we do accept this item, should we change the linked section as part of this PR as well?
Yes, but I would do the bare minimum.
As part of my April snippet cleanup, I was planning on either adding included builds to buildSrc material or flat-out replacing it.